lib/OnyxCache.js

```diff
@@ -0,0 +1,150 @@
+import _ from 'underscore';
+import lodashMerge from 'lodash/merge';
+
+
+const isDefined = _.negate(_.isUndefined);
+
+/**
+ * In memory cache providing data by reference
+ * Encapsulates Onyx cache related functionality
+ */
+class OnyxCache {
+    constructor() {
+        /**
+         * @private
+         * Cache of all the storage keys available in persistent storage
+         * @type {Set<string>}
+         */
+        this.storageKeys = new Set();
+
+        /**
+         * @private
+         * A map of cached values
+         * @type {Record<string, *>}
+         */
+        this.storageMap = {};
```

Comment:
Actually `Map` can be made to work with `lodashMerge` 
I intentionally avoided using a `Map` as I've done so previously (other PR) and was told if I don't have a specific reason to use Map over plain object to prefer a plain object. My specific reason was that Map IMO provides a better interface for mapped data, but as a personal preference it was disregarded 

On the other hand we're using `Set` as it will ignore adding duplicate keys so that field has a specific reason to use modern features

Reply:
When in doubt, let's stick with the status quo.